src/Converter/ShipmentLineItemsConverter.php

```diff
@@ -45,19 +52,45 @@ private function convertUnitRefundToLineItem(ShipmentRefund $shipmentRefund): Li
         ;
         Assert::notNull($shippingAdjustment);
 
+        /** @var ShipmentInterface $shipment */
         $shipment = $shippingAdjustment->getShipment();
         Assert::notNull($shipment);
         Assert::isInstanceOf($shipment, AdjustableInterface::class);
         Assert::lessThanEq($shipmentRefund->total(), $shipment->getAdjustmentsTotal());
 
+        $taxAdjustment = $this->getTaxAdjustment($shipment);
+        $taxAdjustmentAmount = $taxAdjustment !== null ? $taxAdjustment->getAmount() : 0;
+
+        $grossValue = $shipmentRefund->total();
+        $taxAmount = (int) ($grossValue * $taxAdjustmentAmount / $shipment->getAdjustmentsTotal());
```

Comment:
it may be a silly question, but why do we divide by `getAdjustmentsTotal` here?

Reply:
It's identical to `LineItemsConverter` calc, just the way to calculate tax amount part, that corresponds to refund amount out of total unit/shipment amount.